seq_tools/validation/c680_fileSize_fileMd5sum_check.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+
+"""
+    Copyright (c) 2020, Ontario Institute for Cancer Research (OICR).
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+    Authors:
+        Junjun Zhang <junjun.zhang@oicr.on.ca>
+        Linda Xiang <linda.xiang@oicr.on.ca>
+"""
+
+import os
+from collections import defaultdict
+from base_checker import BaseChecker
+from seq_tools.utils import calculate_size, calculate_md5
+
+
+class Checker(BaseChecker):
+    """
+    This checker verify whether fileSize and fileMd5sum provided in metadata are correct or not. 
+    """
+
+    def __init__(self, ctx, metadata):
+        super().__init__(
+            ctx=ctx,
+            metadata=metadata,
+            checker_name=__name__,
+            depends_on=[  # dependent checks
+                'c180_file_uniqueness',
+                'c605_all_files_accessible'
+            ]
+        )
+
+    @BaseChecker._catch_exception
+    def check(self):
+        # status already set at initiation
+        if self.status:
+            return
+
+        files_in_metadata = self.metadata['files']
+
+        mismatches = {}  # dict to keep all mismatches from all files
+        for f in files_in_metadata:
+            seq_file = os.path.join(self.submission_directory, f['fileName'])
+            real_size = calculate_size(seq_file)
+            real_md5 = calculate_md5(seq_file)
+            if mismatches.get(f['fileName']) is None:
+                mismatches[f['fileName']] = []
+            if not real_size == f['fileSize']:
+                mismatches[f['fileName']].append(
+                        "%s: %s vs %s" % ('fileSize', real_size, f['fileSize']))
+            if not real_md5 == f['fileMd5sum']:
+                mismatches[f['fileName']].append(
+                        "%s: %s vs %s" % ('fileMd5sum', real_md5, f['fileMd5sum']))
+            if not mismatches[f['fileName']]: mismatches.pop(f['fileName'])
```

Comment:
not anything major, but the logic seems get a bit complicated here. maybe try `mismatches = defaultdict(list)`, this way, there is not need to have `mismatches[f['fileName']] = []` and no need to pop as well

Reply:
Great point. change it.